### DISCUSSION TITLE
`gunicorn --reload` doesn't work with Uvicorn workers. Bug?

### DISCUSSION THREAD

**Discussion by gwerbin** [Kludex/uvicorn#1638] [Potential Issue] [answered]
I have a trivial FastAPI application in `hello.py`:

```python
from fastapi import FastAPI

app = FastAPI()

@app.get('/')
async def hello():
    # return "hello!"
    return {"msg": "hello!"}
```

I am running this using Gunicorn with Uvicorn workers as follows:

```sh
gunicorn --reload --worker-class 'uvicorn.workers.UvicornWorker' 'hello:app'
```

You can make requests to this server using `curl http://localhost/` and you should see a JSON response.

When I make a change to `hello.py` (e.g. swapping the commented-out lines), the Gunicorn log output indicates that a change was detected and that the server was reloaded. However, checking with `curl` indicates that the app was not actually reloaded; the response does not change.

When I run directly with Uvicorn, reloading works fine:

```sh
uvicorn --reload 'hello:app
```

Is this a Gunicorn bug, a Uvicorn bug, or a "known-bad" interaction?

I am happy to use `uvicorn` directly instead of `gunicorn` for development purposes if this is something that can't be fixed. But I am still curious about this issue.

**Comment by Kludex** [comment, marked as answer]
Kind of a bad already known interaction. Currently, the Uvicorn worker doesn't reload with gunicorn. Use Uvicorn standalone for development.

I'm unable to post references for now, but I have a closed PR with this fix I think.

**Reply by iudeen** [reply]
May be this? https://github.com/encode/uvicorn/pull/1193

**Reply by Kludex** [reply]
Yes

**Reply by gwerbin** [reply]
> Use Unicorn standalone for development.

Good to know and thank you for the quick answer. Maybe this is worth mentioning explicitly in the docs.

**Reply by odravison** [reply]
Has anyone solved this?
